lib/OptimizationManopt/test/runtests.jl

```diff
@@ -105,6 +105,89 @@ end
     @test sol.minimum < 0.1
 end
 
+@testset "CMA-ES" begin
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+
+    opt = OptimizationManopt.CMAESOptimizer()
+
+    optprob = OptimizationFunction(rosenbrock)
+    prob = OptimizationProblem(optprob, x0, p; manifold = R2)
+
+    sol = Optimization.solve(prob, opt)
+    @test sol.minimum < 0.1
+end
+
+@testset "ConvexBundle" begin
+    using QuadraticModels, RipQP
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+
+    opt = OptimizationManopt.ConvexBundleOptimizer()
+
+    optprob = OptimizationFunction(rosenbrock, AutoForwardDiff())
+    prob = OptimizationProblem(optprob, x0, p; manifold = R2)
+
+    sol = Optimization.solve(prob, opt, sub_problem= Manopt.convex_bundle_method_subsolver!) 
+    @test sol.minimum < 0.1
+end
+
+@testset "TruncatedConjugateGradientDescent" begin
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+
+    opt = OptimizationManopt.TruncatedConjugateGradientDescentOptimizer()
+
+    optprob = OptimizationFunction(rosenbrock, AutoForwardDiff())
+    prob = OptimizationProblem(optprob, x0, p; manifold = R2)
+
+    sol = Optimization.solve(prob, opt)
+    @test_broken sol.minimum < 0.1
+end
+
+@testset "AdaptiveRegularizationCubic" begin
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+
+    opt = OptimizationManopt.AdaptiveRegularizationCubicOptimizer()
+
+    optprob = OptimizationFunction(rosenbrock, AutoForwardDiff())
+    prob = OptimizationProblem(optprob, x0, p; manifold = R2)
+
+    @test_broken Optimization.solve(prob, opt)
```

Comment:
NaNs show here as well but looks like the `X` to the hessian function starts becoming NaN, so I am guessing it's a divide by 0 issue somewhere after convergence but the convergence criteria is not triggered. I can use fixed maxiters here and that works but this looks like a bug

```
julia> Optimization.solve(prob, opt, maxiters = 18)
retcode: Default
u: 2-element Vector{Float64}:
 1.0
 1.0

julia> Optimization.solve(prob, opt, maxiters = 19)
ERROR: ArgumentError: invalid argument #4 to LAPACK call
Stacktrace:
  [1] chklapackerror
    @ ~/.julia/juliaup/julia-1.10.0+0.x64.apple.darwin14/share/julia/stdlib/v1.10/LinearAlgebra/src/lapack.jl:38 [inlined]
  [2] gesdd!(job::Char, A::Matrix{Float64})
    @ LinearAlgebra.LAPACK ~/.julia/juliaup/julia-1.10.0+0.x64.apple.darwin14/share/julia/stdlib/v1.10/LinearAlgebra/src/lapack.jl:1668
  [3] svdvals!
    @ ~/.julia/juliaup/julia-1.10.0+0.x64.apple.darwin14/share/julia/stdlib/v1.10/LinearAlgebra/src/svd.jl:217 [inlined]
  [4] svdvals(A::Matrix{Float64})
    @ LinearAlgebra ~/.julia/juliaup/julia-1.10.0+0.x64.apple.darwin14/share/julia/stdlib/v1.10/LinearAlgebra/src/svd.jl:242
  [5] opnorm2(A::Matrix{Float64})
    @ LinearAlgebra ~/.julia/juliaup/julia-1.10.0+0.x64.apple.darwin14/share/julia/stdlib/v1.10/LinearAlgebra/src/generic.jl:679
  [6] opnorm(A::Matrix{Float64}, p::Int64)
    @ LinearAlgebra ~/.julia/juliaup/julia-1.10.0+0.x64.apple.darwin14/share/julia/stdlib/v1.10/LinearAlgebra/src/generic.jl:739
  [7] opnorm(A::Matrix{Float64}, p::Int64)
    @ LinearAlgebra ~/.julia/juliaup/julia-1.10.0+0.x64.apple.darwin14/share/julia/stdlib/v1.10/LinearAlgebra/src/generic.jl:738 [inlined]
  [8] min_cubic_Newton!(mp::DefaultManoptProblem{…}, ls::LanczosState{…}, i::Int64)
    @ Manopt ~/.julia/dev/Manopt/src/solvers/Lanczos.jl:183
  [9] step_solver!(dmp::DefaultManoptProblem{…}, ls::LanczosState{…}, i::Int64)
    @ Manopt ~/.julia/dev/Manopt/src/solvers/Lanczos.jl:167
 [10] solve!(p::DefaultManoptProblem{…}, s::LanczosState{…})
    @ Manopt ~/.julia/dev/Manopt/src/solvers/solver.jl:137
 [11] solve_arc_subproblem!
    @ ~/.julia/dev/Manopt/src/solvers/adaptive_regularization_with_cubics.jl:498 [inlined]
 [12] step_solver!(dmp::DefaultManoptProblem{…}, arcs::AdaptiveRegularizationState{…}, i::Int64)
    @ Manopt ~/.julia/dev/Manopt/src/solvers/adaptive_regularization_with_cubics.jl:470
 [13] step_solver!(amp::DefaultManoptProblem{…}, dss::DebugSolverState{…}, i::Int64)
    @ Manopt ~/.julia/dev/Manopt/src/solvers/debug_solver.jl:26
 [14] step_solver!
    @ ~/.julia/dev/Manopt/src/solvers/solver.jl:150 [inlined]
 [15] solve!(p::DefaultManoptProblem{…}, s::Manopt.ReturnSolverState{…})
    @ Manopt ~/.julia/dev/Manopt/src/solvers/solver.jl:137
 [16] adaptive_regularization_with_cubics!(M::Euclidean{…}, mho::ManifoldHessianObjective{…}, p::Vector{…}; debug::DebugIfEntry{…}, evaluation::InplaceEvaluation, initial_tangent_vector::Vector{…}, maxIterLanczos::Int64, objective_type::Symbol, ρ_regularization::Float64, retraction_method::ExponentialRetraction, σmin::Float64, σ::Float64, η1::Float64, η2::Float64, γ1::Float64, γ2::Float64, θ::Float64, sub_kwargs::@NamedTuple{}, sub_stopping_criterion::StopWhenAny{…}, sub_state::LanczosState{…}, sub_objective::Nothing, sub_problem::Nothing, stopping_criterion::StopWhenAny{…}, kwargs::@Kwargs{…})
    @ Manopt ~/.julia/dev/Manopt/src/solvers/adaptive_regularization_with_cubics.jl:450
 [17] #adaptive_regularization_with_cubics#483
    @ Manopt ~/.julia/dev/Manopt/src/solvers/adaptive_regularization_with_cubics.jl:317 [inlined]
 [18] adaptive_regularization_with_cubics
    @ Manopt ~/.julia/dev/Manopt/src/solvers/adaptive_regularization_with_cubics.jl:313 [inlined]
 [19] #adaptive_regularization_with_cubics#470
    @ Manopt ~/.julia/dev/Manopt/src/solvers/adaptive_regularization_with_cubics.jl:261 [inlined]
 [20] call_manopt_optimizer(M::Euclidean{…}, opt::OptimizationManopt.AdaptiveRegularizationCubicOptimizer, loss::OptimizationManopt.var"#15#16"{…}, gradF::OptimizationManopt.var"#g#17"{…}, x0::Vector{…}; hessF::Function, stopping_criterion::StopWhenAny{…}, evaluation::InplaceEvaluation, retraction_method::ExponentialRetraction, kwargs::@Kwargs{})
    @ OptimizationManopt ~/Optimization.jl/lib/OptimizationManopt/src/OptimizationManopt.jl:284
 [21] __solve(cache::OptimizationCache{…})
    @ OptimizationManopt ~/Optimization.jl/lib/OptimizationManopt/src/OptimizationManopt.jl:470
 [22] solve!(cache::OptimizationCache{…})
    @ SciMLBase ~/.julia/packages/SciMLBase/JUp1I/src/solve.jl:188
 [23] solve(::OptimizationProblem{…}, ::OptimizationManopt.AdaptiveRegularizationCubicOptimizer; kwargs::@Kwargs{…})
    @ SciMLBase ~/.julia/packages/SciMLBase/JUp1I/src/solve.jl:96
 [24] top-level scope
    @ REPL[47]:1
```

Reply:
I can try to find time to check, but I will be travelling on workshops for the next two weeks, so during that time I probably will not find time for this.

Reply:
I've quickly checked this, it looks like a bug. `step_solver!(dmp::AbstractManoptProblem{<:TangentSpace}, ls::LanczosState, i)` with `i == 1` gets a zero-vector in `ls.X` and we have division by 0. I'm not familiar with this solver so I don't know where this case should be handled.

Reply:
Created an issue at Manopt resolving this thread since not supporting second order methods here for now